### PR TITLE
Use new access denied extension point.

### PR DESCRIPTION
**Pull request by adammathys**
Class attribute added in solidusio/solidus#179 allows us to more easily define the behaviour we want for the extension.

### CODE REVIEW

**Comment by gmacdougall** [comment]
:+1:

**Comment by athal7** [comment]
i'm not a huge fan of this, can we find another way to decorate solidus?

**Comment by adammathys** [comment]
@athal7: Updated to hook into Solidus a bit nicer. What do you think?

Tests will fail until solidusio/solidus#179 is merged. Running locally everything is green.

File: lib/controllers/backend/spree/admin/admin_controller_decorator.rb [hunk 1]
```diff
@@ -1,26 +1,11 @@
...
-
   protected
 
-    def model_class
-      const_name = controller_name.classify
-      if Spree.const_defined?(const_name, false)
-        return "Spree::#{const_name}".constantize
-      end
-      nil
+  def model_class
```
Comment (athal7) [inline lib/controllers/backend/spree/admin/admin_controller_decorator.rb]: i'm still curious why this is here as opposed to in core, though that's not central to what you're doing here.

**Comment by athal7** [comment]
looks great!

File: lib/controllers/backend/spree/admin/admin_controller_decorator.rb (hunk 1 above)
Comment (adammathys) [inline lib/controllers/backend/spree/admin/admin_controller_decorator.rb]: Neither am I. Wanted to limit the number of changes in this pull-request so I didn't spend any time looking into whether this is even necessary.

**Comment by jhawthorn** [comment]
:+1:

**Comment by BenMorganIO** [comment]
Writing `rescue_from CanCan::AccessDenied` always felt accessible to me. I can't seem to figure out how to extend this in decorators.

```ruby
# GET /admin/orders

Spree::BaseController.unauthorized_redirect = -> { redirect_to main_app.root_path }

Spee::BaseController.class_eval do
  self.unauthorized_redirect = -> do
    redirect_to main_app.root_path
  end

  def self.unauthorized_redirect
    redirect_to main_app.root_path
  end
end

Spree::Admin::BaseController.unauthorized_redirect = -> { redirect_to main_app.root_path }

Spee::Admin::BaseController.class_eval do
  self.unauthorized_redirect = -> do
    redirect_to main_app.root_path
  end

  def self.unauthorized_redirect
    redirect_to main_app.root_path
  end
end

# => GET /admin/login
```

None of these worked for me, however, using `rescue_from` brought me some desired results.

```
Spree::Admin::BaseController.class_eval do
  rescue_from CanCan::AccessDenied do |_exception|
    render file: 'public/404.html', layout: false, status: 404
  end unless Rails.env.development?
end

# GET /admin
# => 404

# GET /admin/orders
# => 302
```

**Comment by BenMorganIO** [comment]
Not to self and anyone stumbling across via Google, don't forget `self`:

```ruby
Spee::Admin::BaseController.class_eval do
  self.unauthorized_redirect = -> { ... }
end
```

**Pull request merged**